sdk/core/azure-core/src/http/request_activity_policy.cpp

```diff
@@ -23,68 +23,78 @@ std::unique_ptr<RawResponse> RequestActivityPolicy::Send(
     NextHttpPolicy nextPolicy,
     Context const& context) const
 {
-  // Create a tracing span over the HTTP request.
-  std::stringstream ss;
-  // We know that the retry policy MUST be above us in the hierarchy, so ask it for the current
-  // retry count.
-  auto retryCount = RetryPolicy::GetRetryCount(context);
-  if (retryCount == -1)
+  // Find a tracing factory from our context. Note that the factory value is owned by the
+  // context chain so we can manage a raw pointer to the factory.
+  auto tracingFactory = DiagnosticTracingFactory::DiagnosticFactoryFromContext(context);
+  if (tracingFactory)
   {
-    // We don't have a RetryPolicy in the policy stack - just assume this is request 0.
-    retryCount = 0;
-  }
-  ss << "HTTP " << request.GetMethod().ToString() << " #" << retryCount;
-  auto contextAndSpan
-      = Azure::Core::Tracing::_internal::DiagnosticTracingFactory::CreateSpanFromContext(
-          ss.str(), SpanKind::Client, context);
-  auto scope = std::move(contextAndSpan.second);
+    // Create a tracing span over the HTTP request.
+    std::stringstream ss;
```

Comment:
Any particular reason or benefit of using stringstream here for building up the string, rather than just appending/concatenating regular std::string?

Reply:
Originally there were 4 elements being concatenated, one of which was an integer. `stringstream` was the best solution for that case.

Now it's not required but realistically the difference in complexity is probably negligible (adding two strings creates two temporary std::string objects and then assigns their concatenation to a 3rd, using stringstream allocates a buffer, then reallocates the buffer and then creates a std::string from that buffer - probably identical performance characteristics).

Reply:
Makes sense.
It stood out to me because I recall @BillyONeal or @CaseyCarter mention a while back to only use stringstream for formatting and not for string building scenarios like this. But I might be misremembering the detail (or reason).

Reply:
Note that `std::string` does not have the C# usual "append becomes N^2" issue: `const std::string` ~= C# `string`, `std::string` ~= C# `StringBuilder`.

`stringstream` is less efficient in general than using `std::string::append` or `std::string::push_back` directly (where possible to do so). But you can't format numbers or things like that with it.

Reply:
In this example, the two `<<`s are 0 or 1 allocation each (depending on length), then the `str()` is always an extra allocation. If this were instead `auto ss = "HTTP " + request.GetMethod().ToString();`, there is 0 or 1 allocation for the `+` and no second allocation on use.

Reply:
(Note that there's no temporary string constructed for the `"HTTP "`, you call this ctor, assuming `ToString` returns a prvalue:

https://github.com/microsoft/STL/blob/17fde2cbab6e8724d81c9555237c9a623d7fb954/stl/inc/xstring#L5053-L5057

)